doc/changelog.md

```diff
@@ -0,0 +1,919 @@
+-   Add extension to auto-generate function type-hints into documentation.
+    ([SmartSim-PR561](https://github.com/CrayLabs/SmartSim/pull/561))
+-   Add to github release workflow to auto generate a pull request from
+    master into develop for release.
+    ([SmartSim-PR566](https://github.com/CrayLabs/SmartSim/pull/566))
+-   The manifest.json version needs to match the SmartDashboard version,
+    which is 0.0.4 in the upcoming release.
+    ([SmartSim-PR563](https://github.com/CrayLabs/SmartSim/pull/563))
+-   Properly symlinks batch ensembles and batch models.
+    ([SmartSim-PR547](https://github.com/CrayLabs/SmartSim/pull/547))
+-   Remove defensive regexp in .gitignore and ensure tests write to
+    test_output.
+    ([SmartSim-PR560](https://github.com/CrayLabs/SmartSim/pull/560))
+-   After dropping support for Python 3.8, ubuntu needs to be upgraded.
+    ([SmartSim-PR558](https://github.com/CrayLabs/SmartSim/pull/558))
+-   Remove helper function `init_default` and replace with traditional
+    type narrowing.
+    ([SmartSim-PR545](https://github.com/CrayLabs/SmartSim/pull/545))
+-   Ensure the telemetry monitor does not track a task_id for a managed
+    task.
+    ([SmartSim-PR557](https://github.com/CrayLabs/SmartSim/pull/557))
+-   The default path for an entity is now the path to the experiment /
+    the entity name. create_database and create_ensemble now have path
+    arguments. All path arguments are compatible with relative paths.
+    Relative paths are relative to the CWD.
+    ([SmartSim-PR533](https://github.com/CrayLabs/SmartSim/pull/533))
+-   Python 3.8 is reaching its end-of-life in October, 2024, so it will
+    no longer continue to be supported.
+    ([SmartSim-PR544](https://github.com/CrayLabs/SmartSim/pull/544))
+-   Update watchdog dependency from 3.x to 4.x, fix new type issues
+    ([SmartSim-PR540](https://github.com/CrayLabs/SmartSim/pull/540))
+-   The dashboard needs to display historical logs, so log files are
+    written out under the .smartsim directory and files under the
+    experiment directory are symlinked to them.
+    ([SmartSim-PR532](https://github.com/CrayLabs/SmartSim/pull/532))
+-   Add an option to smart build
+    \"\--torch_with_mkl\"/\"\--no_torch_with_mkl\" to prevent Torch from
+    trying to link in the Intel Math Kernel Library. This is needed
+    because on machines that have the Intel compilers installed, the
+    Torch will unconditionally try to link in this library, however
+    fails because the linking flags are incorrect.
+    ([SmartSim-PR538](https://github.com/CrayLabs/SmartSim/pull/538))
+-   Change type_extension and pydantic versions in readthedocs
+    environment to enable docs build.
+    ([SmartSim-PR537](https://github.com/CrayLabs/SmartSim/pull/537))
+-   Promote devices to a dedicated Enum type throughout the SmartSim
+    code base.
+    ([SmartSim-PR498](https://github.com/CrayLabs/SmartSim/pull/498))
```

Comment:
Didn't see this one listed in the generated release docs output!

Reply:
oh hmmm it looks like its in there!

### Miscellaneous Improvements
* Update tutorial containers by @al-rigazzi in https://github.com/CrayLabs/SmartSim/pull/496
* Duplicate entity name prevention by @amandarichardsonn in https://github.com/CrayLabs/SmartSim/pull/480
* Change generic t.any in Experiment API by @mellis13 in https://github.com/CrayLabs/SmartSim/pull/501
* Increase disk space in container for doc builder by @ashao in https://github.com/CrayLabs/SmartSim/pull/504
* Smartsim Documentation Refactor by @amandarichardsonn in https://github.com/CrayLabs/SmartSim/pull/463
* Enforce changelog for SmartSim PRs by @amandarichardsonn in https://github.com/CrayLabs/SmartSim/pull/518
* Remove duplicate launched model names from full test suite by @MattToast in https://github.com/CrayLabs/SmartSim/pull/520
* Mypy: Error on Common Truthy Mistakes by @MattToast in https://github.com/CrayLabs/SmartSim/pull/524
* Add memory & conn collector, collector manager, tests by @ankona in https://github.com/CrayLabs/SmartSim/pull/460
*** Promote Build Device Option to Enum by @amandarichardsonn in https://github.com/CrayLabs/SmartSim/pull/527**
* Disallow Uninitialized Variable Use by @MattToast in https://github.com/CrayLabs/SmartSim/pull/521
* Readthedocs import extension error by @amandarichardsonn in https://github.com/CrayLabs/SmartSim/pull/537
* Enhanced Signal Management by @MattToast in https://github.com/CrayLabs/SmartSim/pull/535
* Update watchdog dependency by @ankona in https://github.com/CrayLabs/SmartSim/pull/540
* Upgrade ubuntu to 22.04 by @AlyssaCote in https://github.com/CrayLabs/SmartSim/pull/558
* Remove defensive regexp in .gitignore by @AlyssaCote in https://github.com/CrayLabs/SmartSim/pull/560
* Fix failing and noisy WLM test by @MattToast in https://github.com/CrayLabs/SmartSim/pull/559
* Bump manifest.json version to 0.0.4 by @AlyssaCote in https://github.com/CrayLabs/SmartSim/pull/563
* Auto-post release PR to develop by @amandarichardsonn in https://github.com/CrayLabs/SmartSim/pull/566
* Auto generate typehints into documentation by @amandarichardsonn in https://github.com/CrayLabs/SmartSim/pull/561
* Set GH_TOKEN environment variable to use Github CLI in workflow by @amandarichardsonn in https://github.com/CrayLabs/SmartSim/pull/570

Reply:
but the PR number was wrong! Correcting